sources/packages/backend/apps/queue-consumers/src/processors/assessment/_tests_/cancel-application-assessment.processor.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+        withDeleted: true,
+      });
+      expect(updatedOveraward.deletedAt).toBeDefined();
+
+      // Assert sent disbursement was not affected.
+      const firstDisbursementUpdated = await disbursementScheduleRepo.findOneBy(
+        { id: firstDisbursement.id },
+      );
+      expect(firstDisbursementUpdated.disbursementScheduleStatus).toBe(
+        DisbursementScheduleStatus.Sent,
+      );
+
+      // Assert pending disbursement was cancelled.
+      const secondDisbursementUpdated =
+        await disbursementScheduleRepo.findOneBy({ id: secondDisbursement.id });
+      expect(secondDisbursementUpdated.disbursementScheduleStatus).toBe(
+        DisbursementScheduleStatus.Cancelled,
+      );
+    });
+
+    it("Should cancel the assessment and log a warning when the workflowInstanceId is not present.", async () => {
+      // Arrange
+      const application = await saveFakeApplicationDisbursements(
+        appDataSource,
+        null,
+        { applicationStatus: ApplicationStatus.Overwritten },
+      );
+      const studentAssessment = application.currentAssessment;
+      // Queued job.
+      const job = createMock<Job<CancelAssessmentQueueInDTO>>({
+        data: { assessmentId: studentAssessment.id },
+      });
+
+      // Act
+      const result = await processor.cancelAssessment(job);
+
+      // Assert
+      expect(zbClientMock.cancelProcessInstance).not.toHaveBeenCalled();
+      expect(result.warnings).toContain(
+        "Assessment was queued to be cancelled but there is no workflow id associated with.",
+      );
+      expect(result.summary).toContain("Assessment cancelled with success.");
+    });
+
+    it("Should throw an error and call job.discard when the application is not in the expected status.", async () => {
+      // Arrange
+      const errorMessage = `Application must be in the ${ApplicationStatus.Cancelled} or ${ApplicationStatus.Overwritten} state to have the assessment cancelled.`;
+      const expectedError = new Error(errorMessage);
+      const application = await saveFakeApplicationDisbursements(
+        appDataSource,
+        null,
+        { applicationStatus: ApplicationStatus.Completed },
+      );
+      const studentAssessment = application.currentAssessment;
+      // Queued job.
+      const job = createMock<Job<CancelAssessmentQueueInDTO>>({
+        data: { assessmentId: studentAssessment.id },
+      });
+
+      // Act and Assert
+      await expect(processor.cancelAssessment(job)).rejects.toStrictEqual(
+        expectedError,
+      );
+      expect(job.discard).toBeCalled();
+    });
+
+    it("Should throw an error and call job.discard when the assessment id was not found.", async () => {
+      // Arrange
+      const assessmentId = 9999;
+      const errorMessage = `Assessment id ${assessmentId} was not found.`;
+      const error = new Error(errorMessage);
+      // Queued job.
+      const job = createMock<Job<CancelAssessmentQueueInDTO>>({
```

Comment:
May or may not be part of the PR, we can have a centralized method to get the jobMock which can have job input type, data and anything else which is required.

Reply:
I started creating it but I reverted it. The `createMock` is already pretty straight forward to be used and the options offered make it pretty easy to use also. If we feel a need in the future we may create a helper, IMO.